lib/spack/llnl/util/filesystem.py

```diff
@@ -2370,7 +2379,23 @@ def add_library_dependent(self, *dest):
                 new_pth = pathlib.Path(pth).parent
             else:
                 new_pth = pathlib.Path(pth)
-            self._additional_library_dependents.add(new_pth)
+            path_is_in_prefix = new_pth.is_relative_to(self.pkg.prefix)
+            add_lib = False
+            if self.link_install_prefix:
+                # We're creating RPath's post install, accept rpath targets anywhere but warn if
+                # out of install prefix
+                if not path_is_in_prefix:
+                    tty.warn(
+                        f"Generating rpath in {new_pth} which is \
+not rooted in target package ({self.pkg}) prefix"
+                    )
+                add_lib = True
+            elif not path_is_in_prefix:
+                # We're creating an RPath for anywhere not in the install prefix, only add rpath
+                # target if it's not in the install prefix
+                add_lib = True
```

Comment:
I'm trying to remember why this happens: is it an accident? i.e. why we would call this for `dest` inside of the prefix, unless someone intentionally placed their test_dir there (if they did we might want to raise an error come to think of it, vs. making this a noop

Reply:
> I'm trying to remember why this happens: is it an accident?

I don't think we had a specific case in mind, just wanted to explicitly handle the two cases we care about. An error could be appropriate, I don't think Spack has populated the install prefix whatsoever by the time the test is invoked, so references to it are likely an error.

Reply:
We could also specifically raise an exception when `dest` doesn't exist. I added a comment that 

Perhaps a more conceptually appropriate refactoring would be to instantiate a `WindowsSimulatedRPath` with a `base_modification_prefix`, and we can only add dependent dirs inside of that prefix so:

* For install RPATHing, we set `base_modification_prefix=Package.prefix`
* For tests, we set `base_modification_prefix=Package.stage.path`

Does that make sense and seem reasonable?